### DISCUSSION TITLE
Provide a way to differentiate between 1 page vs 3 page checkout in function input

### DISCUSSION THREAD

**Discussion by Sam115200** [Shopify/function-examples#281] [API Ideas]
Hi there,

Would it be possible to pass the checkout page location information to the function? For example, in case of 3-page checkout, it can pass this,
Checkout information page -> "/information"
Checkout shipping page -> "/shipping"
Checkout payment page -> "/payment"
<img width="336" alt="Screenshot 2023-07-11 at 5 26 47 PM" src="https://github.com/Shopify/function-examples/assets/139220490/3e57d0b9-c194-424a-8e0c-814120b9b746">

Similarly, for 1-page checkout, it can pass, "/" 
Similar functionality has been requested in [this](https://github.com/Shopify/function-examples/discussions/242#discussion-5204089).

In this way, it would be possible to know the function is running on which checkout page and if this is a 1-page checkout. For certain cases, it is very important to identify and change the function behavior for 1-page as compared to 3-page checkout.

**Comment by lemonmade** [comment]
Can you describe what kinds of changes you would want to make on a 1-page vs 3-page design? We provide the [`buyerJourney.step` field for validation functions](https://shopify.dev/docs/api/functions/reference/cart-checkout-validation/graphql/common-objects/buyerjourney), which tells you some of what you might get out of 1-step vs 3-step (is this validation running on a checkout "completion" or not). We'd ideally like to provide a similarly "semantic" way of describing the information you are after, since the specific page URL is not guaranteed to remain consistent over time.

**Reply by Sam115200** [reply]
@lemonmade Thank you. The [buyerJourney.step](https://shopify.dev/docs/api/functions/reference/cart-checkout-validation/graphql/common-objects/buyerjourney) does sound exactly what I need. Although it seems very limited at the moment, I tested it out and It shows the same step of `CHECKOUT_INTERACTION` on the all the checkout pages, i-e checkout information, shipping and payment page. It only changes to the `CHECKOUT_COMPLETION` when we actually try to `Pay Now` and place the order, which is too late for Payment and Delivery customization functions.

It would be amazing, if you could add additional steps in it, like `CHECKOUT_INFORMATION_INTERACTION`, `CHECKOUT_SHIPPING_INTERACTION`, `CHECKOUT_PAYMENT_INTERACTION` and for one page checkout it could remain `CHECKOUT_INTERACTION`. That way, we will be able to differentiate between 1-page and 3-page checkout.
